core/services/log/registrations.go

```diff
@@ -55,17 +57,18 @@ type (
 
 func newRegistrations() *registrations {
 	return &registrations{
-		registrations: make(map[common.Address]map[common.Hash]map[Listener]*listenerMetadata),
-		decoders:      make(map[common.Address]ParseLogFunc),
+		registrations:          make(map[common.Address]map[common.Hash]map[Listener]*listenerMetadata),
+		decoders:               make(map[common.Address]ParseLogFunc),
+		lowestNumConfirmations: uint64(math.MaxUint64), // a very high value until first subscriber
```

Comment:
Mind adding a little explanation in the comment as to why we need to start high?

Reply:
yup